Dependencies/UrlLib/CMakeLists.txt

```diff
@@ -0,0 +1,35 @@
+if(ANDROID)
+    set(ADDITIONAL_SOURCES
+        "Source/Android/UrlRequest.cpp")
+    set(ADDITIONAL_LIBRARIES
+        PRIVATE AndroidExtensions)
+elseif(APPLE)
+    set(ADDITIONAL_SOURCES
+        "Source/Apple/UrlRequest.mm")
+elseif(WIN32)
+    set(ADDITIONAL_SOURCES
+        "Source/Windows/UrlRequest.cpp")
+    if(NOT WINDOWS_STORE)
+        set(ADDITIONAL_LIBRARIES
+            PUBLIC onecoreuap.lib)
+    endif()
+else()
+    message(FATAL_ERROR "Unrecognized platform: ${CMAKE_SYSTEM_NAME}")
+endif()
+
+set(SOURCES
+    "Include/UrlLib/UrlLib.h"
+    "Source/Shared/UrlRequest.h"
+    ${ADDITIONAL_SOURCES})
+
+add_library(UrlLib ${SOURCES})
+
+target_include_directories(UrlLib PRIVATE "Include")
+target_include_directories(UrlLib PRIVATE "Source")
+target_include_directories(UrlLib INTERFACE "Include")
```

Comment:
nit: Instead of private and interface both pointing to "Include," just one public?

Reply:
fixed